From 012fff9cd7a0f28e99963be5517cb9d6e3a30ec5 Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Fri, 1 Jul 2022 14:43:37 -0700
Subject: [PATCH 1/2] Use find_package for boringssl

The ngtcp2 build seems to allow for multiple backends but in vcpkg we only need one. FindOpenSSL can detect BoringSSL so just use that rather than trying to pass in a list of libraries.
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e709877..406153c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,12 +167,15 @@ endif()
 
 # BoringSSL (required for libngtcp2_crypto_boringssl)
 if(ENABLE_BORINGSSL)
+  find_package(OpenSSL REQUIRED)
+  set(BORINGSSL_INCLUDE_DIR ${OPENSSL_INCLUDE_DIR})
+  set(BORINGSSL_LIBRARIES ${OPENSSL_LIBRARIES})
   cmake_push_check_state()
   set(CMAKE_REQUIRED_INCLUDES   "${BORINGSSL_INCLUDE_DIR}")
   set(CMAKE_REQUIRED_LIBRARIES  "${BORINGSSL_LIBRARIES}")
   check_symbol_exists(SSL_set_quic_early_data_context "openssl/ssl.h" HAVE_SSL_SET_QUIC_EARLY_DATA_CONTEXT)
   if(NOT HAVE_SSL_SET_QUIC_EARLY_DATA_CONTEXT)
-    message(WARNING "Disabling BoringSSL due to lack of QUIC support in ${BORINGSSL_LIBRARIES}")
+    message(FATAL_ERROR "Disabling BoringSSL due to lack of QUIC support in libraries at ${BORINGSSL_LIBRARIES} and incudes at ${BORINGSSL_INCLUDE_DIR}")
   endif()
   cmake_pop_check_state()
 endif()
-- 
2.41.0.windows.1

